### PR TITLE
Downsampling: remove null value setting

### PR DESCRIPTION
**Pull request by weizijun**
The rollup index will have a null value setting, e.g:
```
{
  "downsampling_index" : {
    "settings" : {
      "index.blocks.write" : "true",
      "index.creation_date" : "1662710721497",
      "index.downsample.source.name" : "bench_write",
      "index.downsample.source.uuid" : "eVdGhGKXScezbKWHrxcGzw",
      "index.downsample.status" : "success",
      "index.mapping.total_fields.limit" : null,
      "index.mode" : "time_series",
      "index.number_of_replicas" : "0",
      "index.number_of_shards" : "1",
      "index.provided_name" : "downsampling_index",
      "index.refresh_interval" : "-1",
      "index.routing.allocation.include._tier_preference" : "data_content",
      "index.routing_path" : "labels.*",
      "index.time_series.end_time" : "2016-01-08T00:00:00Z",
      "index.time_series.start_time" : "2016-01-01T00:00:00Z",
      "index.uuid" : "QOw_BASRSoWNG5zc4cl3bw",
      "index.version.created" : "8050099"
    }
  }
}
```

There is a null value settings:
```
      "index.mapping.total_fields.limit" : null,
```

I modify the code and remove it.

### CODE REVIEW

**Comment by pxsalehi** [comment]
@elasticmachine test this please

File: x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/downsample/DownsampleActionSingleNodeTests.java [hunk 1]
```diff
@@ -823,15 +823,22 @@ private void assertRollupIndexSettings(String sourceIndex, String rollupIndex, G
...
             assertNotNull(indexSettingsResp.getSetting(rollupIndex, MapperService.INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING.getKey()));
         }
-        if (indexSettingsResp.getSetting(sourceIndex, MapperService.INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING.getKey()) != null
-            && indexSettingsResp.getSetting(rollupIndex, MapperService.INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING.getKey()) != null)
+
+        if (sourceSettings.keySet().contains(MapperService.INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING.getKey())
+            && rollupSettings.keySet().contains(MapperService.INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING.getKey())) {
             assertEquals(
                 indexSettingsResp.getSetting(sourceIndex, MapperService.INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING.getKey()),
                 indexSettingsResp.getSetting(rollupIndex, MapperService.INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING.getKey())
```
Comment (salvatore-campagna) [inline x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/downsample/DownsampleActionSingleNodeTests.java:836]: Maybe we can use `sourceSettings.get(...)` and `rollupSettings.get(...)` everywhere we use `indexSettingsResp.getSetting(sourceIndex,...)` and indexSettingsResp.getSetting(rollupIndex,...)` here and above?

Comment (weizijun) [reply x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/downsample/DownsampleActionSingleNodeTests.java:836]: done

**Review by salvatore-campagna — APPROVED** [review]

**Comment by salvatore-campagna** [comment]
Thanks for your contribution @weizijun

**Comment by weizijun** [comment]
> Thanks for your contribution @weizijun

Thanks @salvatore-campagna

**Comment by salvatore-campagna** [comment]
@elasticsearchmachine test this please

**Comment by weizijun** [comment]
> Thanks for your contribution @weizijun

Thanks @salvatore-campagna !

**Comment by salvatore-campagna** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.5|[](https://github.com/elastic/elasticsearch/pull/90385)|


### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**